astropy/coordinates/tests/test_masked.py

```diff
@@ -0,0 +1,364 @@
+    def setup_class(self):
+        super().setup_class()
+        self.p = np.linspace(900, 1000, self.msph.size) << u.hPa
+        self.mask_p = np.array([True, False, False, False, False, False])
+        self.mp = Masked(self.p, self.mask_p)
+        # Ensure we have an attribute not associated with the frame.
+        self.fk5 = SkyCoord(self.msph, frame="fk5", pressure=self.mp)
+
+    def test_non_frame_attribute(self):
+        assert_array_equal(self.fk5.get_mask("pressure"), self.mask_p)
+        assert_array_equal(
+            self.fk5.get_mask("pressure", "data"), self.mask | self.mask_p
+        )
+
+
+class TestSkyCoordWithDifferentials:
+    @classmethod
+    def setup_class(self):
+        self.ra = [0.0, 3.0, 6.0, 12.0, 15.0, 18.0] << u.hourangle
+        self.dec = [-15.0, 30.0, 60.0, -60.0, 89.0, -80.0] << u.deg
+        self.dis = [10.0, 20.0, 30.0, 40.0, 50.0, 60.0] << u.pc
+        self.mask_dis = np.array([False, True, False, True, False, True])
+        self.pm_ra_cosdec = [1.0, 2.0, 3.0, -4.0, -5.0, -6.0] << (u.mas / u.yr)
+        self.mask_pm_ra_cosdec = np.array([False, False, True, False, False, True])
+        self.pm_dec = [-9.0, -7.0, 5.0, 3.0, 1.0, 0.0] << (u.mas / u.yr)
+        self.mask_pm_dec = np.array([False, False, True, True, False, True])
+        self.rv = [40.0, 50.0, 0.0, 0.0, -30.0, -10.0] << (u.km / u.s)
+        self.mask_rv = np.array([False, False, False, False, True, True])
+        self.mdis = Masked(self.dis, self.mask_dis)
+        self.mpm_ra_cosdec = Masked(self.pm_ra_cosdec, self.mask_pm_ra_cosdec)
+        self.mpm_dec = Masked(self.pm_dec, self.mask_pm_dec)
+        self.mrv = Masked(self.rv, self.mask_rv)
+        self.sc = SkyCoord(
+            ra=self.ra,
+            dec=self.dec,
+            distance=self.mdis,
+            pm_ra_cosdec=self.mpm_ra_cosdec,
+            pm_dec=self.mpm_dec,
+            radial_velocity=self.mrv,
+        )
+        self.mask = (
+            self.mask_dis | self.mask_pm_ra_cosdec | self.mask_pm_dec | self.mask_rv
+        )
+
+    def test_setup(self):
+        assert self.sc.masked
+        assert_array_equal(self.sc.ra.mask, False)
+        assert_array_equal(self.sc.dec.mask, False)
+        assert_array_equal(self.sc.distance.mask, self.mask_dis)
+        assert_array_equal(self.sc.pm_ra_cosdec.mask, self.mask_pm_ra_cosdec)
+        assert_array_equal(self.sc.pm_dec.mask, self.mask_pm_dec)
+        assert_array_equal(self.sc.radial_velocity.mask, self.mask_rv)
+        assert_array_equal(self.sc.mask, self.mask)
+
+    def test_get_mask(self):
+        assert_array_equal(self.sc.get_mask(), self.mask)
+        assert_array_equal(self.sc.get_mask("ra", "dec"), False)
+
+    @pytest.mark.parametrize(
+        "dt",
+        [
+            1 * u.yr,
+            Masked([1, 2, 3] * u.yr, mask=[False, True, False])[:, np.newaxis],
+        ],
+    )
+    def test_apply_space_motion(self, dt):
+        sc = self.sc.apply_space_motion(dt=dt)
```

Comment:
There should be tests for unmasking the propagated coordinates.

Reply:
Done